src/cli.rs

```diff
@@ -64,7 +64,7 @@ pub(crate) struct Args {
 
     // options for --each-feature and --feature-powerset
     /// --optional-deps [DEPS]...
-    pub(crate) optional_deps: Option<Vec<String>>,
```

Comment:
`[DEPS]` is optional, so `Some(vec![])` and `None` have different meaning:

- `--optional-deps` means "use *all* optional deps as features"
- `--optional-deps val1,val2,..`/`--optional-deps=val1,val2,..` means "use the specified optional deps val1, val2, and .. as features"

And special `--optional-deps=` (not `--optional-deps`) behaves as if `--optional-deps` is not specified, but IIRC this was the syntax for CI to allow writing as follows:

```yml
      matrix:
        include:
          - rust: 1.70 # msrv
            optional-deps: '' 
          - rust: stable
            optional-deps: 'deps-needs-latest-stable'
      # ...
      run: cargo hack check --optional-deps=${{ matrix.optional-deps }}
```

Reply:
(`--optional-deps=` is also [tested](https://github.com/taiki-e/cargo-hack/blob/d0121f596bbf8aefad6252119ce462be3b4ffc73/tests/test.rs#L1235), but probably not documented anywhere, I should have written a document about it)